### PR TITLE
Fix config

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] tidy-up
- [x] backfill tests (this project has 0 tests)
  - [x] unit
  - [x] acceptance
    - [x] no engines
    - [x] engines
    - [ ] ~~external eyeglass dependencies~~
- [x] confirm in big app 
  - [x] tested little app \w engines
  - [x] big app with \w engines

Release Plan:

1. release v3.3.x with this (which includes deprecation and transition path)
2. release v4.0.0 with the deprecation and transition path removed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,11 +11,29 @@ Then rename all your `.css` files so they have the `.scss` extension.
...
+const app = new EmberApp(defaults, {
+  eyeglass: { /* configuration */ }
+});
+```
+
+### Engines
+
+```js
+// path/to/engine/index.js
+const app = new EmberEngine(defaults, {
```
Comment (rwjblue) [inline README.md]: This isn't valid syntax.  Most likely you mean:

```js
module.exports = EngineAddon.extend({
  options: {
    eyeglass: { /* configuration */ }
  }
});
```

File: index.js [hunk 2]
```diff
@@ -111,13 +104,25 @@ module.exports = {
...
+          // TODO: WTF engines
+          let from = addon.parent.root + '/config/environment';
+          let to = addon.parent.root + '/ember-cli-build';
+          addon.ui.writeDeprecateLine(`'eyeglass' configuration within config/environment is no longer supported\n  please move this configuration:\n\tfrom: '${from}' (or however configured) \n\tto:   '${to}' (or however configured)\n`);
+          config = cloneDeep(projectConfig.eyeglass);
+        } else {
+          // setup eyeglass for this project's configuration
+          const hostConfig = cloneDeep(host.options.eyeglass || {});
+          const addonConfig = isNestedAddon ? cloneDeep(addon.parent.options.eyeglass || {}) : {};
+          config = defaultsDeep(addonConfig, hostConfig);
```
Comment (rwjblue) [inline index.js]: This seems unrelated to migrating from `config/environment.js` to me (though the feature itself seems nice)...

File: package.json [hunk 3]
```diff
@@ -53,12 +53,10 @@
...
     "broccoli-merge-trees": "^2.0.0",
-    "lodash.clonedeep": "^4.5.0"
+    "lodash.clonedeep": "^4.5.0",
+    "lodash.defaultsdeep": "^4.6.0"
   },
   "ember-addon": {
     "configPath": "tests/dummy/config"
-  },
-  "engines": {
-    "node": "^4.5 || 6.* || >= 7.*"
```
Comment (rwjblue) [inline package.json]: Why remove engines config?

File: README.md (hunk 1 above)
Comment (stefanpenner) [reply README.md]: correct

File: index.js (hunk 2 above)
Comment (stefanpenner) [reply index.js]: should i drop?

File: package.json (hunk 3 above)
Comment (stefanpenner) [reply package.json]: rebase fail

Comment (stefanpenner) [reply package.json]: actually not sure whats up,   
```
"engines": {
    "node": "^6.* || >= 7.*"
  },
```

is in the code

File: tests/acceptance/eyeglass-test.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import QUnit from 'qunit';
+
+QUnit.assert.contains = function(needle, haystack, message) {
+  const actual = haystack.indexOf(needle) > -1;
+  this.push(actual, actual, needle, message);
+};
+
+moduleForAcceptance('Acceptance | eyeglass');
+
+test('visiting /', function(assert) {
```
Comment (stefanpenner) [inline tests/acceptance/eyeglass-test.js]: @rwjblue example of expectance tests.

**Comment by stefanpenner** [comment]
failing test is expected, as lazy-engines + ember-cli-eyeglass appears to be in a brittle state. I'll likely need to:

* find the exact configuration that works today
* leave some skipped tests for future cleanup work

File: index.js [hunk 5]
```diff
@@ -79,90 +80,46 @@ function localEyeglassAddons(addon) {
...
 
+    debugger;
     registry.add('css', {
       name: 'eyeglass',
       ext: 'scss',
       toTree(tree, inputPath, outputPath) {
-        let host = findHost(addon);
-        let inApp = (host === addon.app);
-
+        debugger;
```
Comment (stefanpenner) [inline index.js]: ^^

File: .travis.yml [hunk 6]
```diff
@@ -21,7 +21,6 @@ env:
   matrix:
     # we recommend new addons test the current and previous LTS
     # as well as latest stable release (bonus points to beta/canary)
-    - EMBER_TRY_SCENARIO=ember-lts-2.8
```
Comment (rwjblue) [inline .travis.yml]: Discussed with Stef RE: why this is removed. This was not supported by ember-engines.

File: index.js [hunk 7]
```diff
@@ -79,90 +80,46 @@ function localEyeglassAddons(addon) {
...
-    if (env != 'production') {
-      defaults.eyeglass.verbose = false
-    }
-    return defaults;
-  },
-
   setupPreprocessorRegistry(type, registry) {
     let addon = this;
 
+    debugger;
```
Comment (rwjblue) [inline index.js]: Should be removed.

File: index.js [hunk 8]
```diff
@@ -183,3 +140,72 @@ module.exports = {
...
+
+  let config;
+  if (projectConfig.eyeglass) {
+    // TODO: WTF engines
+    let from = addon.parent.root + '/config/environment';
+    let to = addon.parent.root + '/ember-cli-build';
+    addon.ui.writeDeprecateLine(`'eyeglass' configuration within config/environment is no longer supported\n  please move this configuration:\n\tfrom: '${from}' (or however configured) \n\tto:   '${to}' (or however configured)\n`);
+    config = cloneDeep(projectConfig.eyeglass);
+  } else {
+    const isNestedAddon = typeof addon.parent.parent === 'object';
```
Comment (stefanpenner) [inline index.js]: functional change is this branch

File: index.js [hunk 9]
```diff
@@ -183,3 +140,72 @@ module.exports = {
...
+  if (addon.parent && addon.parent.engineConfig) {
+    projectConfig = addon.parent.engineConfig(host.env, projectConfig);
+  }
+
+  let config;
+  if (projectConfig.eyeglass) {
+    // TODO: WTF engines
+    let from = addon.parent.root + '/config/environment';
+    let to = addon.parent.root + '/ember-cli-build';
+    addon.ui.writeDeprecateLine(`'eyeglass' configuration within config/environment is no longer supported\n  please move this configuration:\n\tfrom: '${from}' (or however configured) \n\tto:   '${to}' (or however configured)\n`);
```
Comment (stefanpenner) [inline index.js]: Lets but this behind `process.env.EYEGLASS_DEPRECATE_ON_BROWSER_CONFIG`

**Comment by stefanpenner** [comment]
released as v3.4.0 🎉
